### ISSUE TITLE
Enable protein mutation transformations for amino acids with nonstandard protonation states

### ISSUE THREAD

**Issue by zhang-ivy** [choderalab/perses#831]
Since we are failing to achieve convergence for K27A and D39A in barnase:barstar, @jchodera suggested that I run these mutations with alternate protonation states (LYN instead of LYS and ASH instead of ASP). If we can demonstrate that the DDG is sensitive to the protonation state, then we can say that our lack of convergence for these mutations could likely be solved with constant pH simulations. 

Currently, the `PointMutationEngine`/`PolymerProposalEngine` do not support nonstandard amino acids. When constructing the atom map, the PolymerProposalEngine relies on generating an oemol from one of the PDBs in [perses/data/amino_acid_templates](https://github.com/choderalab/perses/tree/master/perses/data/amino_acid_templates).  The code for this is [here](https://github.com/choderalab/perses/blob/222e1c7003bcd5b8a8e0c9f28396301e5e2bdaaf/perses/rjmc/topology_proposal.py#L1228-L1232)

To enable protein mutation transformations involving LYN and ASH, is is sufficient to add PDBs for LYN and ASH in `amino_acid_templates`? If so, how were these PDBs generated?
Are there other changes (to topology_proposal.py or geometry.py) necessary to handle these types of amino acids?

Tagging @dominicrufa  and @jchodera  for feedback!

**Comment by dominicrufa** [comment]
> To enable protein mutation transformations involving LYN and ASH, is is sufficient to add PDBs for LYN and ASH in amino_acid_templates? If so, how were these PDBs generated?

this was awhile ago, but i think i just pulled the amino acid templates from `pymol` and use the standard `openeye.oechem`  MCSS map to generate mappings (ignoring the hydrogen caps at the ends of the acid).

**Comment by jchodera** [comment]
We should consider using the [PDB Chemical Component Dictionary](https://www.wwpdb.org/data/ccd) which contains a dictionary of chemical components and protonation variants of amino acids with unique names. This page provide mmCIF files, but SDF/MOL files for all these protonation state variants are available at the [Ligand Expo](). For example, here's [`GLN_LL`](http://ligand-expo.rcsb.org/pyapps/ldHandler.py?formid=cc-index-search&target=GLN_LL&operation=ccid). You can download the whole thing [here](http://ligand-expo.rcsb.org/ld-download.html).

I think @ijpulidos has been working on ways in which we could auto-perceive or automatically access these templates in the RCSB. I imagine we can leverage this information to automatically generate transformations between amino acids (or nonnatural amino acids) and protonation state variants by deriving constrained mappings from amino acids in the Ligand Expo.

**Comment by zhang-ivy** [comment]
>We should consider using the PDB Chemical Component Dictionary 

Yes, I looked into using this, but got confused because it didn't seem like the protonation states were correct for the amino acids I looked at. For example, `LYN`, which is deprotonated LYS, should contain a sidechain with a terminal NH2, not NH3, but the CCD entry [here](https://www.ebi.ac.uk/pdbe-srv/pdbechem/chemicalCompound/show/LYN)  contains NH3 as the terminal group. This could be because `LYN` as defined by Amber has a different name in the CCD, but I decided it wasn't worth my effort to look more into this. 

> I imagine we can leverage this information to automatically generate transformations between amino acids (or nonnatural amino acids) and protonation state variants by deriving constrained mappings from amino acids in the Ligand Expo.

The path of least resistance for me to be able to run experiments nonstandard protonation state amino acids involves the changes in #834 . I still have to fix one test, but after this PR is merged, we can keep this issue open so that we remember to overhaul the current `_construct_atom_map()` code with something more generalizable to ALL nonstandard amino acids (not just the 6 that are covered in my PR. 

Since this would be an enhancement, this seems like it would be a low priority item for @ijpulidos to tackle?
